### ISSUE TITLE
Interpreter used for new Virtual Environment

### ISSUE THREAD

**Issue by DataWookie** [rstudio/reticulate#294]
Hi!

Firstly, I'm loving working with `reticulate`!

I am, however, having some trouble creating virtual environments which use Python 3.

For example:

```
> virtualenv_create("test")
Creating virtualenv at  /home/colliera/.virtualenv/test 
New python executable in /home/colliera/.virtualenv/test/bin/python
Installing setuptools, pip, wheel...done.
```

The new virtual environment is created with the Python 2 interpreter (although I have Python 3 on my system and reticulate is actually running with Python 3 too).

```
> py_config()
python:         /usr/bin/python3
libpython:      /usr/lib/python3.6/config-3.6m-x86_64-linux-gnu/libpython3.6.so
pythonhome:     /usr:/usr
version:        3.6.5 (default, Apr  1 2018, 05:46:30)  [GCC 7.3.0]
numpy:          /home/colliera/.local/lib/python3.6/site-packages/numpy
numpy_version:  1.14.5

python versions found: 
 /usr/bin/python
 /usr/bin/python3
```

I can see why this is happening:

```
> reticulate:::virtualenv_config()
$python
[1] "/usr/bin/python"

$virtualenv
[1] "/usr/local/bin/virtualenv"

$pip_version
[1] "pip"

$root
[1] "/home/colliera/.virtualenv"
```

So the virtual environment configuration is set up to use Python 2. Digging into the code for `virtualenv_config()` I see that it will always pick `python_unix_binary("python")` in preference to `python_unix_binary("python")`.

Would it not make more sense to provide a way to specify which executable should be used for the virtual environment? Or, maybe, default to the version that is being used already by `reticulate`?

I'd be very happy to submit a pull request, but before I do that I just wanted to understand whether there's reasoning behind the present implementation.

Thanks,
Andrew.

**Comment by jjallaire** [comment]
You are correct that we currently prefer Python 2 to Python 3. This is mostly because just about every Linux distribution (until very recent ones like Ubuntu 18.04) shipped with Python 2 by default as did/does OS X. Our goal originally was to try to use "system" Python as often as possible and this aligned with that goal.

In more recent development versions of reticulate we will detect and use Python 3 but only when there is no Python 2. We could definitely change this to make it more flexible/configurable, but I don't want to modify that codepath right now b/c we are in a test cycle prior to the next CRAN release and that code unfortunately gets a bit complicated.
